### PR TITLE
SAK-31659 check password for null before setting it

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/UserEntityProvider.java [hunk 1]
```diff
@@ -241,7 +241,9 @@ public void updateEntity(EntityReference ref, Object entity, Map<String, Object>
             edit.setEmail(u.getEmail());
             edit.setFirstName(u.getFirstName());
             edit.setLastName(u.getLastName());
-            edit.setPassword(u.getPassword());
+            if (u.getPassword() != null) {
```
Comment (steveswinsburg) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/UserEntityProvider.java]: Is "" or "            " ok?
Perhaps StringUtils.isEmpty/isBlank?

Comment (steveswinsburg) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/UserEntityProvider.java]: Also, createEntity has a similar line, does that need an update?
`edit.setPassword(user.getPassword());`

Comment (ottenhoff) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/UserEntityProvider.java]: I don't think createEntity needs fixing .... the problem I'm trying to solve is that updating a user property nulls out their password ....

**Comment by ottenhoff** [comment]
Okay I added another commit to check for blank password also

**Pull request merged**